platform-operator/controllers/verrazzano/component/opensearchoperator/opensearchoperator.go

```
@@ -0,0 +1,498 @@
+		resourceClient := client.Resource(gvr)
+		objList, err := resourceClient.Namespace(ComponentNamespace).List(context.TODO(), metav1.ListOptions{})
+		if err != nil {
+			return err
+		}
+		if len(objList.Items) > 0 {
+			return fmt.Errorf("waiting for all %s to be deleted", gvr.String())
+		}
+	}
+	return nil
+}
+
+func getActualCRNodes(cr *vzapi.Verrazzano) map[string]vzapi.OpenSearchNode {
+	nodeMap := map[string]vzapi.OpenSearchNode{}
+	if cr != nil && cr.Spec.Components.Elasticsearch != nil {
+		for _, node := range cr.Spec.Components.Elasticsearch.Nodes {
+			nodeMap[node.Name] = node
+		}
+	}
+	return nodeMap
+}
+
+func getMasterNode(ctx spi.ComponentContext) string {
+	for _, node := range ctx.EffectiveCR().Spec.Components.Elasticsearch.Nodes {
+		for _, role := range node.Roles {
+			if node.Replicas == nil || *node.Replicas < 1 {
+				continue
+			}
+			if role == "master" {
+				return node.Name
+			}
+		}
+	}
+	return ""
+}
+
+// buildIngressOverrides builds the overrides required for the OpenSearch and OpenSearchDashboards ingresses
+func buildIngressOverrides(ctx spi.ComponentContext, kvs []bom.KeyValue) ([]bom.KeyValue, error) {
+	if vzcr.IsNGINXEnabled(ctx.EffectiveCR()) {
+
+		dnsSubDomain, err := vzconfig.BuildDNSDomain(ctx.Client(), ctx.EffectiveCR())
+		if err != nil {
+			return kvs, ctx.Log().ErrorfNewErr("Failed to build DNS subdomain: %v", err)
+		}
+		ingressClassName := vzconfig.GetIngressClassName(ctx.EffectiveCR())
+		ingressTarget := fmt.Sprintf("verrazzano-ingress.%s", dnsSubDomain)
+
+		ingressAnnotations := make(map[string]string)
+		ingressAnnotations[`cert-manager\.io/cluster-issuer`] = constants.VerrazzanoClusterIssuerName
+		if vzcr.IsExternalDNSEnabled(ctx.EffectiveCR()) {
+			ingressAnnotations[`external-dns\.alpha\.kubernetes\.io/target`] = ingressTarget
+			ingressAnnotations[`external-dns\.alpha\.kubernetes\.io/ttl`] = "60"
+		}
+
+		kvs, _ = appendOSIngressOverrides(ingressAnnotations, dnsSubDomain, ingressClassName, kvs)
```

why are the errors being ignored?

---

It was not really returning any error since we I was just appending values to kvs